python/cudf/cudf/core/column/column.py

```diff
@@ -474,9 +518,13 @@ def __setitem__(self, key, value):
         from cudf.core import column
 
         if isinstance(key, slice):
-            key_start, key_stop, key_stride = key.indices(len(self))
-            if key_stride != 1:
+            if key.step:
```

Comment:
we need to handle `key.step == 1` and `key.step is None` here, this will throw if `key.step == 1` currently.